### PR TITLE
[DO NOT MERGE] RelAPI Onboarding 

### PR DESCRIPTION
**Pull request by claire-labry**
### Description

These are the necessary components that `terraform-provider-aws` needs to onboard to Releases API. 

edit: rel-eng will merge this PR when the time is ready.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -30,25 +30,50 @@ jobs:
...
-      setup-signore-github-token: '${{ secrets.HASHI_SIGNORE_GITHUB_TOKEN }}'
-      signore-client-id: '${{ secrets.SIGNORE_CLIENT_ID }}'
-      signore-client-secret: '${{ secrets.SIGNORE_CLIENT_SECRET }}'
-    with:
-      goreleaser-release-args: --timeout 2h
-      hc-releases-aws-role-duration-seconds: 7200
-      release-notes: true
-      setup-go-version: '${{ needs.go-version.outputs.version }}'
+          
+      - name: Setup hc-releases
```
[the remark this answers is not in the document]
Comment (claire-labry) [reply .github/workflows/release.yml:34]: Hi there @anGie44 -- great question! @jeanneryan has this PR for ghaction-terraform-provider-release which should align with what you're asking: https://github.com/hashicorp/ghaction-terraform-provider-release/pull/8/commits/c10cda4838fcdbc693ad101227ed2845ca01374b

[the remark this answers is not in the document]
Comment (claire-labry) [reply .github/workflows/release.yml:34]: Oh I think I misunderstood your ask -- brb, doing some testing to see how to work this out + will update the PR accordingly.

[the remark this answers is not in the document]
Comment (claire-labry) [reply .github/workflows/release.yml]: @anGie44 -- thank you for calling this out! I've updated the PR to reflect your ask / suggestions :)

[the remark this answers is not in the document]
Comment (anGie44) [reply .github/workflows/release.yml]: Ahhh I didn't see that PR already in the works 😅 !  thanks for the updates @claire-labry , looking forward to using the new API 🥳

**Review by jeanneryan — APPROVED** [review]

**Comment by anGie44** [comment]
@claire-labry , @jeanneryan: I just rebased to fix conflicts with recent changes in `release.yml`

**Pull request merged**